bundles/aero.minova.rcp.rcp/src/aero/minova/rcp/rcp/fields/NumberField.java

```diff
@@ -26,47 +32,122 @@
 import org.eclipse.swt.widgets.Control;
 import org.eclipse.swt.widgets.Label;
 import org.eclipse.swt.widgets.Text;
+import org.eclipse.swt.widgets.ToolTip;
 
 import aero.minova.rcp.constants.Constants;
 import aero.minova.rcp.css.CssData;
 import aero.minova.rcp.css.CssType;
+import aero.minova.rcp.model.Value;
 import aero.minova.rcp.model.form.MNumberField;
+import aero.minova.rcp.model.util.NumberFormatUtil;
 import aero.minova.rcp.rcp.accessor.NumberValueAccessor;
+import aero.minova.rcp.util.OSUtil;
 
 public class NumberField {
 
-	public static Control create(Composite composite, MNumberField field, int row, int column, Locale locale, MPerspective perspective) {
+	public static Control create(Composite composite, MNumberField field, int row, int column, Locale locale, MPerspective perspective,
+			TranslationService translationService) {
 		String unitText = field.getUnitText() == null ? "" : field.getUnitText();
 
 		Label label = FieldLabel.create(composite, field);
-		Text text = TextFactory.newText(SWT.BORDER | SWT.RIGHT).text("").create(composite);
-		NumberValueAccessor numberValueAccessor = new NumberValueAccessor(field, text);
 
-		text.addFocusListener(new FocusAdapter() {
+		TextAssistContentProvider contentProvider = new TextAssistContentProvider() {
+
 			@Override
-			public void focusGained(FocusEvent e) {
-				text.selectAll();
+			public List<String> getContent(String entry) {
+				ArrayList<String> result = new ArrayList<>();
+				int decimals = field.getDecimals();
+
+				NumberFormat numberFormat = NumberFormat.getNumberInstance(locale);
+				numberFormat.setMaximumFractionDigits(decimals);
+				numberFormat.setMinimumFractionDigits(decimals);
+				numberFormat.setGroupingUsed(true);
```

Comment:
Das würde ich nicht jedes mal in der getContent Methode neu erstellen, sondern einmal in der überliegenden Methode